enterprise/models.py

```diff
@@ -541,3 +541,81 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+@python_2_unicode_compatible
+class EnterpriseCourseEnrollment(TimeStampedModel):
+    """
+    Store information about the enrollment of a user in a course.
+
+    This model is the central source of truth for information about
+    whether a particular user, linked to a particular EnterpriseCustomer,
+    has been enrolled in a course, and is the repository for any other
+    relevant metadata about such an enrollment.
+    """
+
+    class Meta(object):
+        unique_together = (('user', 'course_id',),)
+        app_label = 'enterprise'
+
+    user = models.ForeignKey(
```

Comment:
I think we need to be careful calling this a 'user' -- it's not actually a User it's an EnterpriseCustomerUser, which also is not a User, it's a relationship between an Enterprise and a User.